business/istio_validations.go

```diff
@@ -265,13 +265,17 @@ func (in *IstioValidationsService) fetchNonLocalmTLSConfigs(mtlsDetails *kuberne
 		return
 	}
 
-	meshPolicies, err := in.k8s.GetMeshPolicies(config.Get().IstioNamespace)
-	if err != nil {
-		errChan <- err
-		return
-	} else {
-		mtlsDetails.MeshPolicies = meshPolicies
-	}
+	mpChan := make(chan []kubernetes.IstioObject)
+	mpErrChan := make(chan error)
+	go func() {
```

Comment:
You forgot ``wg.Add(1)`` and ``defer wg.Done()``. Otherwise this goroutine might not be finished when the results are checked. Also, these channels do not work this way as you can't block with ``wg.Wait()`` here. So reading channels later for errors is not possible.

So instead, create pointers that you will associate (and send them to the go func). It's enough to check the error inside the goroutine and only then associate the pointer.

Reply:
Like this:

https://play.golang.org/p/XqAmIoxPH1w

Reply:
You can safely ignore the wg.Wait() from that example in your code, but the example won't work without it.

Reply:
oh, I see. You take advantage of the `wg` WaitingGroup you already have as a parameter.